coqlspclient/proof_file.py

```diff
@@ -158,53 +161,67 @@ def get_context(file_path: str, timeout: int):
                 v_file = aux_file.get_diagnostics(
                     "Locate Library", library, last_line + i + 1
                 ).split()[-1][:-1]
-                coq_file = CoqFile(v_file, library=library, timeout=timeout)
-                coq_file.run()
+
+                with open(v_file, "r") as f:
+                    hash = hashlib.md5(f.read().encode("utf-8")).hexdigest()
```

Comment:
Why store a hash? Is this related with the fact that we can now change files (add/remove steps)?

Reply:
If a file changes during the run, we want to know that the file has to be analyzed again. The changes can be done by adding or removing steps, but may also be external to the program.